src/main/java/core/basesyntax/utils/generate/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.utils.generate.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.generate.ReportGenerator;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final ReportGenerator generator;
+
+    public ReportGeneratorImpl(ReportGenerator generator) {
+        this.generator = generator;
+    }
+
+    @Override
+    public String getReport(StorageImpl storage) {
+        return generator.getReport(storage);
+    }
```

Comment:
The `getReport` method should not delegate to another `ReportGenerator`. Instead, it should implement the logic to generate the report based on the data from `StorageImpl`. Please refactor this method to generate the report directly.